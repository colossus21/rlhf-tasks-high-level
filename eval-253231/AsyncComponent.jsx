
import React, { useState, useEffect } from 'react';

function AsyncComponent() {
  const [data, setData] = useState(null);
  useEffect(() => {
    fetchData().then((result) => {
      setData(result);
    });
  }, []);
  const fetchData = async () => {
    const response = await fetch('https://randomuser.me/api/?results=10');
    const result = await response.json();
    return result;
  };
  return <div>{data ? data.message : 'Loading...'}</div>;
}

export default AsyncComponent;
    