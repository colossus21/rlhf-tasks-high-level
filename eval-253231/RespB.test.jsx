import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import AsyncComponent from './AsyncComponent';

beforeAll(() => {
    // Mock the fetch function before running tests
    global.fetch = jest.fn(() =>
        Promise.resolve({
            json: () => Promise.resolve({ message: 'Success' }),
        })
    );
});

afterAll(() => {
    // Clean up fetch mock after tests
    global.fetch.mockClear();
});

test('renders loading message initially', () => {
    const { container } = render(<AsyncComponent />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Loading...')).toBeInTheDocument();
});

test('renders fetched data correctly', async () => {
    const { container } = render(<AsyncComponent />);

    // Wait for the async data to load
    await waitFor(() => expect(screen.getByText('Success')).toBeInTheDocument());

    expect(container).toMatchSnapshot();
});