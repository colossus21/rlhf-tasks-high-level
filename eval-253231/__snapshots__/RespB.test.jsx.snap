// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders fetched data correctly 1`] = `
<div>
  <div>
    Success
  </div>
</div>
`;

exports[`renders loading message initially 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
